dev-util/android-sdk-update-manager/android-sdk-update-manager-26.1.1.ebuild

```diff
@@ -0,0 +1,109 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+PYTHON_COMPAT=(python3_{10..11})
+
+inherit python-r1 udev distutils-r1
+
+DESCRIPTION="Open Handset Alliance's Android SDK"
+HOMEPAGE="https://developer.android.com"
+SRC_URI="https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip"
+RESTRICT="mirror"
+
+LICENSE="android"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="app-arch/tar
+		app-arch/unzip
+		app-arch/gzip
+		acct-group/android"
+
+RDEPEND=">=virtual/jdk-1.5
+	>=dev-java/ant-core-1.6.5
+	dev-java/swt:3.7[cairo]
+	sys-libs/ncurses-compat:5[abi_x86_32(-)]
+	sys-libs/zlib[abi_x86_32(-)]
+"
+
+ANDROID_SDK_DIR="/opt/${PN}"
+QA_FLAGS_IGNORED_x86="
+	${ANDROID_SDK_DIR/\//}/tools/adb
+	${ANDROID_SDK_DIR/\//}/tools/mksdcard
+	${ANDROID_SDK_DIR/\//}/tools/sqlite3
+	${ANDROID_SDK_DIR/\//}/tools/hprof-conv
+	${ANDROID_SDK_DIR/\//}/tools/zipalign
+	${ANDROID_SDK_DIR/\//}/tools/dmtracedump
+"
+QA_FLAGS_IGNORED_amd64="${QA_FLAGS_IGNORED_x86}"
+
+QA_PREBUILT="${ANDROID_SDK_DIR/\//}/tools/*"
+
+S="${WORKDIR}/tools"
+
+src_compile() {
+    :
+}
```

Comment:
Is that really required? If you don't have anything to do inside src_compile it should be enough just to drop it

Reply:
When I deleted it I got error:
```
>>> Compiling source in /var/tmp/portage/dev-util/android-sdk-update-manager-26.1.1/work/tools ...
 * python3_10: running distutils-r1_run_phase distutils-r1_python_compile
 * ERROR: dev-util/android-sdk-update-manager-26.1.1::gentoo failed (compile phase):
 *   esetup.py: setup.py nor setup.cfg not found
 *
 * Call stack:
 *     ebuild.sh, line  122:  Called src_compile
 *   environment, line 3370:  Called distutils-r1_src_compile
 *   environment, line 1547:  Called _distutils-r1_run_foreach_impl 'distutils-r1_python_compile'
 *   environment, line  683:  Called python_foreach_impl 'distutils-r1_run_phase' 'distutils-r1_python_compile'
 *   environment, line 3066:  Called multibuild_foreach_variant '_python_multibuild_wrapper' 'distutils-r1_run_phase' 'distutils-r1_python_compile'
 *   environment, line 2617:  Called _multibuild_run '_python_multibuild_wrapper' 'distutils-r1_run_phase' 'distutils-r1_python_compile'
 *   environment, line 2615:  Called _python_multibuild_wrapper 'distutils-r1_run_phase' 'distutils-r1_python_compile'
 *   environment, line 1010:  Called distutils-r1_run_phase 'distutils-r1_python_compile'
 *   environment, line 1529:  Called distutils-r1_python_compile
 *   environment, line 1338:  Called esetup.py 'build' '-j' '8'
 *   environment, line 2135:  Called die
 * The specific snippet of code:
 *               die "${FUNCNAME}: setup.py nor setup.cfg not found";
 *
 * If you need support, post the output of `emerge --info '=dev-util/android-sdk-update-manager-26.1.1::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=dev-util/android-sdk-update-manager-26.1.1::gentoo'`.
 * The complete build log is located at '/var/tmp/portage/dev-util/android-sdk-update-manager-26.1.1/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/dev-util/android-sdk-update-manager-26.1.1/temp/environment'.
 * Working directory: '/var/tmp/portage/dev-util/android-sdk-update-manager-26.1.1/work/tools'
 * S: '/var/tmp/portage/dev-util/android-sdk-update-manager-26.1.1/work/tools'
 ```
 So I stay with it.

Reply:
And to elaborate: If left unspecified, it will run the eclass/EAPI defaults - as you saw, it may not **always** work but in 99 % of time it does.